### PR TITLE
Lab 06

### PR DESCRIPTION
**Pull request by DashaDrahun**
Problems:
1)    in public class StartUp:    
 public IIdentityService CreateIdentityService()
        {
            return new IdentityService(new UnitOfWorkUserIdentity(new ClientManager(new IdentityContext())));
        }
2) in public class AutofacBlConfig:
            builder.RegisterType<IdentityContext>().InstancePerRequest();
     in public class UnitOfWorkPassengersCarriageFunc:
 public UnitOfWorkPassengersCarriageFunc(**PassengersContext db**, IRepository<Route> routeRepository, IRepository<Trip> tripRepository, IRepository<Order> orderRepository)
        {
            **this.db = db;** // not new assengersContext()
            this.routeRepository = routeRepository;
            this.tripRepository = tripRepository;
            this.orderRepository = orderRepository;
        }

### CODE REVIEW

**Comment by DimaViaroukin** [comment]
Please remove empty folder ViewModelsForViews -> RegisteredUser and rename ViewModelsForViews to ViewModels

File: Lab06.MVC/Lab06.MVC.Carriage/Global.asax.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+{
+    public class MvcApplication : System.Web.HttpApplication
+    {
+        protected void Application_Start()
+        {
+            var builder = new ContainerBuilder();
+            builder.RegisterControllers(typeof(MvcApplication).Assembly).PropertiesAutowired();
+            builder.RegisterModule(new AutofacPlConfig());
+
+            IMapper mapper = new MapperConfiguration(cfg =>
```
Comment (DimaViaroukin) [inline Lab06.MVC/Lab06.MVC.Carriage/Global.asax.cs]: It should be as method for initialization for Autofac
For example Autofac.Inin() or Autofac.BuildMapping or something like this

File: Lab06.MVC/Lab06.MVC.Carriage.BL/Interfaces/IUserService.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
```
Comment (DimaViaroukin) [inline Lab06.MVC/Lab06.MVC.Carriage.BL/Interfaces/IUserService.cs]: please remove unused usings

File: Lab06.MVC/Lab06.MVC.Carriage.BL/Model/RouteModel.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿namespace Lab06.MVC.Carriage.BL.Model
+{
+    public class RouteModel
+    {
+        public int RouteId { get; set; }
```
Comment (DimaViaroukin) [inline Lab06.MVC/Lab06.MVC.Carriage.BL/Model/RouteModel.cs]: please remove unused properties

File: Lab06.MVC/Lab06.MVC.Carriage.BL/Model/TripModel.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+﻿namespace Lab06.MVC.Carriage.BL.Model
+{
+    public class TripModel
+    {
+        public int TripId { get; set; }
+        public int RouteId { get; set; }
```
Comment (DimaViaroukin) [inline Lab06.MVC/Lab06.MVC.Carriage.BL/Model/TripModel.cs]: Please remove unused properties

File: Lab06.MVC/Lab06.MVC.Carriage.DAL/Interfaces/IUnitofWorkUserIdentity.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Threading.Tasks;
+using Lab06.MVC.Carriage.DAL.Identity;
+
+namespace Lab06.MVC.Carriage.DAL.Interfaces
+{
+    public interface IUnitofWorkUserIdentity:IDisposable
```
Comment (DimaViaroukin) [inline Lab06.MVC/Lab06.MVC.Carriage.DAL/Interfaces/IUnitofWorkUserIdentity.cs]: please format this class and fix name IUnitofWorkUserIdentity to IUnit**O**fWorkUserIdentity

File: Lab06.MVC/Lab06.MVC.Carriage.DAL/PassengersContextMigrations/Configuration.cs [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    {
+        public Configuration()
+        {
+            AutomaticMigrationsEnabled = false;
+            MigrationsDirectory = @"PassengersContextMigrations";
+        }
+
+        protected override void Seed(Context.PassengersContext context)
+        {
+            // seeding by initial data - in migration code
```
Comment (DimaViaroukin) [inline Lab06.MVC/Lab06.MVC.Carriage.DAL/PassengersContextMigrations/Configuration.cs]: Should it be empty?

**Comment by DimaViaroukin** [comment]
image
please fix packages (all projects should have the same versions)

File: .gitignore [hunk 7]
```diff
@@ -8,4 +8,8 @@
 .vs
 bin
 obj
-packages
\ No newline at end of file
+packages
+*/App_Start/Web.Debug.config
+*/App_Start/Web.Release.config
+*/Web.Debug.config
```
Comment (DimaViaroukin) [inline .gitignore]: it should be in git

File: .gitignore [hunk 8]
```diff
@@ -8,4 +8,8 @@
...
+*/Web.Release.config
```
Comment (DimaViaroukin) [inline .gitignore]: it should be in git

File: Lab06.MVC/Readme.md [hunk 9]
```diff
@@ -0,0 +1,7 @@
+1) Create database from Package Manager by commands:
+ Update-Database -configuration Lab06.MVC.Carriage.DAL.IdentityContextMigrations.Configuration �Verbose
```
Comment (DimaViaroukin) [inline Lab06.MVC/Readme.md]: please fix special symbol in readme

File: Lab06.MVC/Lab06.MVC.Carriage/Controllers/AccountController.cs [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+                }
+                else
+                {
+                    AuthenticationManager.SignOut();
+                    AuthenticationManager.SignIn(
+                        new AuthenticationProperties
+                    {
+                        IsPersistent = false
+                    },
+                        claim);
```
[the remark this answers is not in the document]
Comment (DashaDrahun) [reply Lab06.MVC/Lab06.MVC.Carriage/Controllers/AccountController.cs]: SonarQube requires such formatting

File: Lab06.MVC/Lab06.MVC.Carriage.BL/Model/TripModel.cs (hunk 4 above)
Comment (DashaDrahun) [reply Lab06.MVC/Lab06.MVC.Carriage.BL/Model/TripModel.cs]: These properties are used for viewing routes and updating trips in View

**Pull request merged**